marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/writer/WriteDataWithOptionsTest.java

```diff
@@ -6,13 +6,102 @@
 import com.marklogic.hub.spark.sql.sources.v2.AbstractSparkConnectorTest;
 import org.junit.jupiter.api.Test;
 
+import org.apache.spark.sql.sources.v2.writer.WriterCommitMessage;
+
 import java.util.Set;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
 
 public class WriteDataWithOptionsTest extends AbstractSparkConnectorTest {
 
+    @Test
```

Comment:
This test class already has too many scenarios in it - let's make a new "WriteDataWithUriTemplateTest" to capture all of these tests. Makes it easy to see all the test coverage for this feature.

Reply:
ready for commit